internal/resources/otelcollector/core.go

```diff
@@ -21,18 +21,35 @@ import (
 )
 
 // applyCommonResources applies resources to gateway and agent deployment node
-func applyCommonResources(ctx context.Context, c client.Client, name types.NamespacedName, clusterRole *rbacv1.ClusterRole, allowedPorts []int32) error {
-	// Create RBAC resources in the following order: service account, cluster role, cluster role binding.
+func applyCommonResources(ctx context.Context, c client.Client, name types.NamespacedName, rbac rbac, allowedPorts []int32) error {
+	// Create service account before RBAC resources
 	if err := k8sutils.CreateOrUpdateServiceAccount(ctx, c, makeServiceAccount(name)); err != nil {
 		return fmt.Errorf("failed to create service account: %w", err)
 	}
 
-	if err := k8sutils.CreateOrUpdateClusterRole(ctx, c, clusterRole); err != nil {
-		return fmt.Errorf("failed to create cluster role: %w", err)
+	// Create RBAC resources in the following order: cluster role, cluster role binding, role, role binding
+	if rbac.clusterRole != nil {
+		if err := k8sutils.CreateOrUpdateClusterRole(ctx, c, rbac.clusterRole.DeepCopy()); err != nil {
```

Comment:
can you please write a comment why using DeepCopy

Reply:
done 👍🏻